action-owm.py

```diff
@@ -166,39 +166,38 @@ def searchWeatherForecastItem(hermes, intent_message):
     region = getRegion(intent_message)
     country = getCountry(intent_message)
     geographical_poi = getPOI(intent_message)
-    res = hermes.skill.speak_item(item_name,
+    res = hermes.skill.speak_condition(item_name,
                                   datetime,
                                   granularity=granularity,
-                                 Locality=locality,
-                                 Region=region,
-                                 Country=country,
-                                 POI=geographical_poi)
+                                  Locality=locality,
+                                  Region=region,
+                                  Country=country,
+                                  POI=geographical_poi)
     current_session_id = intent_message.session_id
     hermes.publish_end_session(current_session_id, res.decode("latin-1"))
 
 if __name__ == "__main__":
     config = read_configuration_file("config.ini")
 
-    if config.get("secret").get("api_key") is None:
-        print "No API key in config.ini, you must setup an OpenWeatherMap API key for this skill to work"
-    elif len(config.get("secret").get("api_key")) == 0:
-        print "No API key in config.ini, you must setup an OpenWeatherMap API key for this skill to work"
-    
-    skill_locale = config.get("secret", {"locale":"en_US"}).get("locale", u"en_US")
-    
-    if skill_locale == u"":
-        print "No locale information is found!"
-        print "Please edit 'config.ini' file, give either en_US, fr_FR or es_ES refering to the language of your assistant"
+    api_key = config.get("secret", {}).get("api_key")
+    if not api_key:
+        print("No API key in config.ini, you must setup an OpenWeatherMap API key for this skill to work")
         sys.exit(1)
-        
-    skill = SnipsOWM(config["secret"]["api_key"],
-            config["secret"]["default_location"],locale=skill_locale.decode('ascii'))
     
+    skill_locale = config["secret"].get("locale")
+    if not skill_locale:
+        print("No locale information is found!")
+        print("Please edit 'config.ini' file, give either en_US, fr_FR or es_ES refering to the language of your assistant")
+        sys.exit(1)
+
+    # TODO config["secret"]["default_location"] can be empty is this intended?
+    skill = SnipsOWM(api_key, config["secret"]["default_location"], locale=skill_locale.decode('ascii'))
+
```

Comment:
is it intended that default_location can be empty here? (thats tested for all other parameters)